### DISCUSSION TITLE
Better controller support for Playstation controllers

### DISCUSSION THREAD

**Discussion by waspennator** [ikemen-engine/Ikemen-GO#2010] [Feature Request]
### Is your feature request related to a problem? Please describe.

My Xbox series X controller seems to work alright with Ikemen GO, so I decided to try out my Dualsense and Dualshock 4 controllers and noticed some weird things.

1: For some reason the Taunt button and pause button are assigned to the right trigger and left trigger respectively.
2: The D-Pad does not move my character around, only the analog stick seems to do it.

My Xbox Series X controller exhibits none of these issues, the taunt and pause buttons are assigned to the start and select buttons respectively and the D-pad works fine on there.

### Describe the solution you'd like

It'd be cool if the game could know if a Playstation controller was plugged in and adjusted the taunt button, pause button, and the D-pad accordingly to work like my Xbox controller.

### Additional context

Products tested:

Dualshock 4 Gen 1 Controller
Dualsense  Controller
Xbox Series X Controller

### Search terms

Controller Support

**Comment by Jesuszilla** [comment]
DualSense support has just been added, please test. I need to find a Dualshock 4 to test with (used to have one, cannot find it now) but my new roommate ought to have one so I'll address that soon.

Also, what OS are you using? Windows, Linux, macOS? It DOES matter. Also, can you run I.K.E.M.E.N-Go from command line, go to joystick config for the proper index, and give me any output from the terminal? Ideally make sure no other controllers are plugged in or paired so it always goes to P1 and that's a surefire way of knowing the index. If it works, then you were on the wrong index; if it doesn't, then there ought to be output. macOS currently has an issue where joystick configs will be swapped because joystick indices are not consistent across program starts, keep in mind. Also, is it being used over Bluetooth or USB? That may also matter as far as controller names go.

Also @K4thos can this be converted to an issue?
